File: 7단계/1157.java
import java.util.*;

public class Main{
    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);
        String input = sc.next();
        int arr[] = new int[26];
        int max=0;
        char result='?';
        
        for(int i=0; i<input.length(); i++){
            if('A' <= input.charAt(i) && input.charAt(i) <= 'Z'){
                arr[input.charAt(i) - 'A']++;
            }else
                arr[input.charAt(i) - 'a']++;
        }
        for(int i=0; i<26; i++){
            if(arr[i]>max){
                max=arr[i];
                result = (char)(i+'A');
            }
            else if(arr[i]==max){
                result = '?';
            }
        }
        System.out.println(result);
    }
}

File: 7단계/1157.py
st = input().upper()#문자를 대문자로 받음 MISSISSIPI


word=list(set(st))#list타입으로 중복을 제거해서 저장(문자마다 카운트하기 위함) M,I,P,S
print(st)
print(word)
arr=[]
for i in word:
    arr.arrppend(st.count(i))#st에 i(word)가 몇개 있는지 카운트해서 arr리스트에 저장
print(arr)
if arr.count(max(arr)) >= 2:
    print("?")
else:
    print(word[(arr.index(max(arr)))])
